fullscreen/api/element-request-fullscreen-active-document.html

```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<title>Element#requestFullscreen() when the document is not the active document</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<iframe allowfullscreen></iframe>
+<script>
+var t = async_test();
+
+onload = t.step_func(() => {
+  var iframe = document.querySelector("iframe");
+  var documentBeforeNav = iframe.contentDocument;
+  documentBeforeNav.onfullscreenchange = t.unreached_func('fullscreenchange event');
+  documentBeforeNav.onfullscreenerror = t.unreached_func('fullscreenerror event');
+
+  iframe.onload = t.step_func(() => {
+    var p = documentBeforeNav.documentElement.requestFullscreen();
+    // If a promise was returned, it should already be rejected, so its reject
+    // callback should be invoked before a second promise's callback.
+    // This test does not fail if promises aren't implemented.
+    if (p) {
+      var rejected = false;
+      p.catch(t.step_func(() => rejected = true));
+      Promise.resolve().then(t.step_func(() => assert_true(rejected)));
+    }
+    // Wait two animation frames to ensure that no events are fired. (One should
+    // be enough per spec, but this makes the test flaky in Firefox.)
```

Comment:
I wasn't concerned with any transition because this is an error case, rather I want to wait just long enough to know that the fullscreenerror event is not fired. Per current spec that should be at the next animation frame, so one rAF should be enough.

I looked into Gecko's code and see that you use `AsyncEventDispatcher`, which looks like the equivalent of "queue a task to fire an event". I've confirmed that a single setTimeout would make the test fail reliably in Firefox, and that setting the timeout just before the requestFullscreen call makes it pass reliably, so that checks out.

I'll tweak the test to use a mix of rAF and setTimeout to handle either implementation.